backend/src/dao/TeamEventAttendanceDao.ts

```diff
@@ -51,4 +52,59 @@ export default class TeamEventAttendanceDao {
     const docRef = teamEventAttendanceCollection.doc(uuid);
     await docRef.delete();
   }
+
+  /**
+   * Deletes all TEC Attendance
+   */
+  static async deleteAllTeamEventAttendance(): Promise<void> {
+    const batch = teamEventAttendanceCollection.firestore.batch();
+    const coll = await teamEventAttendanceCollection.get();
+
+    coll.docs.forEach((doc) => batch.delete(doc.ref));
+    await batch.commit();
+  }
+
+  /**
+   * Gets all TEC Attendance for a user
+   * @param user - user whose attendance should be fetched
+   */
+  static async getTeamEventAttendanceByUser(user: IdolMember): Promise<TeamEventAttendance[]> {
+    const memberRef = memberCollection.doc(user.email);
+    const refs = await teamEventAttendanceCollection.where('member', '==', memberRef).get();
+    const attendance = refs.docs.map((doc) => doc.data() as DBTeamEventAttendance);
+    return Promise.all(
+      attendance.map(async (att) => ({
+        ...att,
+        member: await getMemberFromDocumentReference(att.member)
+      }))
+    );
+  }
+
+  /**
+   * Gets all TEC Attendance for a given team event
+   * @param uuid - DB uuid of team event
+   */
+  static async getTeamEventAttendanceByEventId(uuid: string): Promise<TeamEventAttendance[]> {
+    const refs = await teamEventAttendanceCollection.where('eventUuid', '==', uuid).get();
+    const attendance = refs.docs.map((doc) => doc.data() as DBTeamEventAttendance);
+    return Promise.all(
+      attendance.map(async (att) => ({
+        ...att,
+        member: await getMemberFromDocumentReference(att.member)
+      }))
+    );
+  }
+
+  /**
+   * Gets all TEC Attendance for all events
+   */
+  static async getAllTeamEventAttendance(): Promise<TeamEventAttendance[]> {
+    const attendanceRefs = await teamEventAttendanceCollection.get();
+    return Promise.all(
+      attendanceRefs.docs.map(async (doc) => ({
+        ...doc.data(),
+        member: await getMemberFromDocumentReference(doc.get('member'))
```

Comment:
I don't think we need to call `get` from the doc every time. `doc.data()` should have the member information. I believe your logic here would double the amount of reads for each member (which we've been working to reduce these days). Since it's not too complicated to optimize here I would add that: save doc.data() in a variable (of type `DBTeamEventAttendance`) and then grab the `member` field off of that.